### PR TITLE
Reduce Debug Console Cpu Overhead with Condition Variables

### PR DESCRIPTION
**Pull request by kylekrol**
### Summary of Changes

* Removed the lockless queue in favor of a deque and condition variables from the stl.
* This also prevents each flight software binary from pinning a Cpu core in Hootl which will hopefully help with performance issues with our big integrated testing campaign.

### Testing

Timing performance with updates:
mutex

and without updates:
mutexless

I don't see any noticeable regressions?

### CODE REVIEW

File: src/fsw/FCCode/DebugTask.cpp [hunk 1]
```diff
@@ -20,12 +20,20 @@ void DebugTask::execute() {
 #ifndef FLIGHT
   start_cycle_f.set(false);
   if(auto_cycle_f.get()){
-    process_commands(_registry);
+    process_commands(_registry
+#ifdef DESKTOP
+    , false
+#endif
+    );
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/DebugTask.cpp:27]:
```suggestion
    #ifdef DESKTOP
       const bool blocking = true;
    #else
       const bool blocking = false;
    #endif
    process_commands(_registry, blocking);
```

Just a little cleaner, IMO

**Review by tanishqaggarwal — APPROVED** [review]
Gud shit

File: src/fsw/FCCode/DebugTask.cpp (hunk 1 above)
Comment (kylekrol) [reply src/fsw/FCCode/DebugTask.cpp:27]: Yeah no good move for sure!

Comment (kylekrol) [reply src/fsw/FCCode/DebugTask.cpp]: Oh wait actually can't do this because the argument doesn't exist in HOOTL.

Comment (tanishqaggarwal) [reply src/fsw/FCCode/DebugTask.cpp]: Oh why not just let the argument exist and be useless in HITL...oh the compiler will complain?

Comment (tanishqaggarwal) [reply src/fsw/FCCode/DebugTask.cpp]: Whatever it doesn't matter

Comment (kylekrol) [reply src/fsw/FCCode/DebugTask.cpp]: Compiler errors for unused argument yeah.

**Pull request merged**